### ISSUE TITLE
Property mis-identified as static member of same class when bound in constructor

### ISSUE THREAD

**Issue by RevanProdigalKnight** [microsoft/TypeScript#35024]
**TypeScript Version:**  3.7.2

**Search Terms:** "static member of"

**Code**

```ts
class ProgressBar extends HTMLElement {
  private static updateBar(this: ProgressBar, pct: string) {
    /* etc */
  }

  public constructor() {
    super();
        
    this.updateBar = ProgressBar.updateBar.bind(this);
  }
}
```

**Expected behavior:** No compilation errors

**Actual behavior:** The assignment to `updateBar` in the constructor (and all other usages) incorrectly throws `error TS2576: Property 'updateBar' is a static member of type 'ProgressBar'`

**Playground Link:** http://www.typescriptlang.org/play/?esModuleInterop=false&ts=3.8.0-dev.20191105&ssl=1&ssc=1&pln=11&pc=2#code/MYGwhgzhAEAKBOB7A5vAplAQme00A8AXNAOwBMYAJAFQFkAZAURDQFtTDoBvAKGn+gAHeAEsAbmGLQIhSSODQAroLKS02eAApCACxEQAXHCSoMEDQBohwQkZmiSyAJTc+A9wHoAVHkIKvHjwAkAC+PMGCigBGIPLQwIgk9oo2iFouvO7uEMpo6QDcblnQwUG6+gB0yqrEGtAAvMYo6Fg4VSpqGhVRIuTaehBOhaE8IUA

**Related Issues:** None that I could find

**Comment by fatcerberus** [comment]
Merely assigning `this.x` in the constructor isn't enough to create an instance property.  You can see this if you add `this.foo = 0;` to the constructor:

> Property 'foo' does not exist on type 'ProgressBar'.

This, for example, fixes the error:

```ts
class ProgressBar extends HTMLElement {
	private static updateBar(this: ProgressBar, pct: string) {
		/* etc */
	}

	updateBar: typeof ProgressBar.updateBar;

	public constructor() {
		super();

		this.updateBar = ProgressBar.updateBar.bind(this);
	}
}
```

**Comment by RevanProdigalKnight** [comment]
I could've sworn I tried that and it didn't work... but now it does.
